### PR TITLE
Add users

### PR DESCRIPTION
**Pull request by nbuonin**

### CODE REVIEW

File: mailchimp_subscriber.py [hunk 1]
```diff
@@ -1,7 +1,53 @@
...
+import csv
 from mailchimp3 import MailChimp
 
+EMAIL_RE = re.compile(r'(^[a-zA-Z0-9_+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)')
+
+
+def validate_email(email_address):
+    """Validate the syntax of the email address"""
+    match = EMAIL_RE.match(email_address)
+    if match:
```
Comment (sdreher) [inline mailchimp_subscriber.py]: simpler: `return match is not None`

File: mailchimp_subscriber.py [hunk 2]
```diff
@@ -1,7 +1,53 @@
...
+    """Client is a representation of a CTL client"""
+    def __new__(cls, row):
+        if (len(row) >= 3 and
+                validate_email(row[0])):
+            return super(Client, cls).__new__(cls)
+        else:
+            raise ValueError
+
+    def __init__(self, row):
+        self.email_address = row[0].strip()
```
Comment (sdreher) [inline mailchimp_subscriber.py]: replace hard-coded numbers (a.k.a magic numbers) with constant definitions. Something like FIRST_NAME_ROW = 1.

File: mailchimp_subscriber.py [hunk 3]
```diff
@@ -15,21 +61,39 @@ def load_conf(conf_file):
...
     to the MailChimp list"""
-    client = MailChimp(mc_user, mc_key)
-    contacts = client.lists.members.all(list_id, get_all=True,
-                                        status='subscribed',
-                                        fields="members.email_address")
-    email_addresses = {contact['email_address']
-                       for contact in contacts['members']}
-    return users.difference(email_addresses)
+    mc_client = MailChimp(mc_user, mc_key)
+    for client in users.values:
```
Comment (sdreher) [inline mailchimp_subscriber.py:84]: Why iterate the users.values list twice here?

File: mailchimp_subscriber.py [hunk 4]
```diff
@@ -15,21 +61,39 @@ def load_conf(conf_file):
...
+        set_mailchimp_status(client, mc_client, list_id)
+    for client in users.values:
+        add_user(client, mc_client, list_id)
+
+
+def add_user(client, mc_client, list_id):
+    if (client.mailchimp_status == 'pending' or
```
Comment (sdreher) [inline mailchimp_subscriber.py:90]: Should this conditional include `== 'pending'`?

Comment (nbuonin) [reply mailchimp_subscriber.py:90]: Yes, I think so.  A client could have one of six different statuses.  Only in these two cases do I want it to act on them.  Is there a better way to test for these two?

Comment (sdreher) [reply mailchimp_subscriber.py:90]: Asking as I think the code here tries to update the status to "pending" from "not present". If the user's status is already "pending", then this operation seems redundant.

File: mailchimp_subscriber.py [hunk 5]
```diff
@@ -1,7 +1,54 @@
...
+def validate_email(email_address):
+    """Validate the syntax of the email address"""
+    match = EMAIL_RE.match(email_address)
+    return match is not None
+
+
+class Client:
+    """Client is a representation of a CTL client"""
+    def __new__(cls, row):
+        if (len(row) >= 3 and
```
Comment (sdreher) [inline mailchimp_subscriber.py:25]: swap in your constants here and at Ln. 26 as well.

**Pull request merged**